web-client/src/templates/Task.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+import Link from "next/link";
+
+type TaskViewerProps = {
+  id: number;
+  title: string;
+  slug: string;
+};
+
+const TaskViewer = (props: TaskViewerProps) => {
+  return (
+    <div>
+      <Link href={`/tasks/view/?idOrSlug=${props.id}`}>
```

Comment:
Can we use this `/tasks/:id/:slug` url instead? It's much friendlier for the browser so people can just type the problem name directly into the url bar and it'll autocomplete better.

`/tasks/${props.id}/${props.slug}`